http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -354,6 +406,122 @@ class RuntimeServiceInterp[F[_]: Parallel](blocker: Blocker,
+                                                         runtime: Runtime,
+                                                         runtimeConfig: RuntimeConfig,
+                                                         traceId: TraceId): F[Unit] =
+    (runtimeConfig, req) match {
+      case (RuntimeConfig.GceConfig(existingMachineType, existingDiskSize),
+            UpdateRuntimeConfigRequest.GceConfig(reqMachineType, reqDiskSize)) =>
+        for {
+          // should machine type be updated?
+          updatedMachineType <- reqMachineType.flatTraverse[F, (MachineTypeName, Boolean)] { mt =>
+            if (mt != existingMachineType)
+              if (runtime.status == RuntimeStatus.Stopped)
+                Async[F].pure(Some((mt, false)))
+              else if (!allowStop)
+                Async[F].raiseError(RuntimeMachineTypeCannotBeChangedException(runtime))
+              else Async[F].pure(Some((mt, true)))
+            else Async[F].pure(None)
+          }
+          // should disk size be updated?
+          updatedDiskSize <- reqDiskSize.flatTraverse[F, Int] {
+            case d if d < existingDiskSize =>
+              Async[F].raiseError(RuntimeDiskSizeCannotBeDecreasedException(runtime))
+            case d if d > runtimeConfig.diskSize =>
+              Async[F].pure(Some(d))
+            case _ =>
+              Async[F].pure(None)
+          }
+          // if either of the above is defined, send a PubSub message
+          _ <- (updatedMachineType orElse updatedDiskSize).traverse_ { _ =>
+            val message = UpdateRuntimeMessage(runtime.id,
+                                               updatedMachineType.map(_._1),
+                                               updatedMachineType.map(_._2).getOrElse(false),
+                                               updatedDiskSize,
+                                               None,
+                                               None,
+                                               Some(traceId))
+            publisherQueue.enqueue1(message)
+          }
+        } yield ()
+
+      case (RuntimeConfig.DataprocConfig(existingNumWorkers,
+                                         existingMasterMachineType,
+                                         existingMasterDiskSize,
+                                         _,
+                                         _,
+                                         _,
+                                         existingNumPreemptibles,
+                                         _),
+            UpdateRuntimeConfigRequest.DataprocConfig(reqMasterMachineType,
+                                                      reqMasterDiskSize,
+                                                      reqNumOfWorkers,
+                                                      reqNumPreemptibles)) =>
+        for {
+          // should num workers be updated?
+          updatedNumWorkers <- reqNumOfWorkers.flatTraverse[F, Int] {
```

Comment:
minor rename `updated` to `target`? `updated` reads to me as if the update has already happened